cmd/abigen/main.go

```diff
@@ -0,0 +1,278 @@
+			if err != nil {
+				utils.Fatalf("Failed to build Solidity contract: %v", err)
+			}
+		case c.GlobalIsSet(vyFlag.Name):
+			output, err := compiler.CompileVyper(c.GlobalString(vyperFlag.Name), c.GlobalString(vyFlag.Name))
+			if err != nil {
+				utils.Fatalf("Failed to build Vyper contract: %v", err)
+			}
+			contracts = make(map[string]*compiler.Contract)
+			for n, contract := range output {
+				name := n
+				// Sanitize the combined json names to match the
+				// format expected by solidity.
+				if !strings.Contains(n, ":") {
+					// Remove extra path components
+					name = abi.ToCamelCase(strings.TrimSuffix(filepath.Base(name), ".vy"))
+				}
+				contracts[name] = contract
+			}
+
+		case c.GlobalIsSet(jsonFlag.Name):
+			jsonOutput, err := ioutil.ReadFile(c.GlobalString(jsonFlag.Name))
+			if err != nil {
+				utils.Fatalf("Failed to read combined-json from compiler: %v", err)
+			}
+			contracts, err = compiler.ParseCombinedJSON(jsonOutput, "", "", "", "")
+			if err != nil {
+				utils.Fatalf("Failed to read contract information from json output: %v", err)
+			}
+		}
+		// Gather all non-excluded contract for binding
+		for name, contract := range contracts {
+			if exclude[strings.ToLower(name)] {
+				continue
+			}
+			abi, err := json.Marshal(contract.Info.AbiDefinition) // Flatten the compiler parse
+			if err != nil {
+				utils.Fatalf("Failed to parse ABIs from compiler output: %v", err)
+			}
+			abis = append(abis, string(abi))
+			bins = append(bins, contract.Code)
+			sigs = append(sigs, contract.Hashes)
+			nameParts := strings.Split(name, ":")
+			types = append(types, nameParts[len(nameParts)-1])
+
+			libPattern := crypto.Keccak256Hash([]byte(name)).String()[2:36]
+			libs[libPattern] = nameParts[len(nameParts)-1]
+		}
+	}
+	// Extract all aliases from the flags
+	if c.GlobalIsSet(aliasFlag.Name) {
+		// We support multi-versions for aliasing
+		// e.g.
+		//      foo=bar,foo2=bar2
+		//      foo:bar,foo2:bar2
+		re := regexp.MustCompile(`(?:(\w+)[:=](\w+))`)
+		submatches := re.FindAllStringSubmatch(c.GlobalString(aliasFlag.Name), -1)
+		for _, match := range submatches {
+			aliases[match[1]] = match[2]
+		}
+	}
+	// Generate the contract binding
+	code, err := bind.Bind(types, abis, bins, sigs, c.GlobalString(pkgFlag.Name), lang, libs, aliases)
+	if err != nil {
+		utils.Fatalf("Failed to generate ABI binding: %v", err)
+	}
+	// Either flush it out to a file or display on the standard output
+	if !c.GlobalIsSet(outFlag.Name) {
+		fmt.Printf("%s\n", code)
+		return nil
+	}
+	if err := ioutil.WriteFile(c.GlobalString(outFlag.Name), []byte(code), 0600); err != nil {
+		utils.Fatalf("Failed to write ABI binding: %v", err)
+	}
+	return nil
+}
+
+func main() {
+	log.Root().SetHandler(log.LvlFilterHandler(log.LvlInfo, log.StreamHandler(os.Stderr, log.TerminalFormat(true))))
+
+	if err := app.Run(os.Args); err != nil {
```

Comment:
what is this file for?

Reply:
this creates the binary for `abigen` in `env GOBIN= go install ./cmd/abigen` (line 175)

Reply:
q: is it possible to replace it with the go code below?

Reply:
the go code below (the link you had sent in below comment) is generated by `abigen`. So will need this to deploy contract through go

Reply:
or maybe use `geth abigen`? I'm sceptic of adding a dependency that we will have to maintain and constantly check if it's not deprecated.

Reply:
done with commit 45d88f3